desc/utils.py

```diff
@@ -684,4 +692,66 @@ def broadcast_tree(tree_in, tree_out, dtype=int):
         raise ValueError("trees must be nested lists of dicts")
 
 
+@partial(jnp.vectorize, signature="(m),(m)->(n)", excluded={"size", "fill_value"})
+def take_mask(a, mask, /, *, size=None, fill_value=None):
```

Comment:
Isn't regular mask jit compialible?

Reply:
No;  `a[mask]` returns only the elements of `a` where `mask` is true. When the number of true elements in `mask` is unknown, this is a dynamic array size operation, which is not JIT compatible or differentiable using the JAX AD tool.